### PR TITLE
Default to last command in external editor.

### PR DESCRIPTION
**Pull request by tsroten**
## Description

This pull request mimics some of the behavior in the official mysql client.
- When you type `\e` without a query in front of it, mycli use the last-run query as the default text in the editor.
- If you type `\e` and there were no queries run before it, then the editor will be blank.

This addresses https://github.com/dbcli/mycli/issues/35

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: mycli/main.py [hunk 1]
```diff
@@ -419,8 +419,9 @@ def handle_editor_command(self, cli, document):
         saved_callables = cli.application.pre_run_callables
         while special.editor_command(document.text):
             filename = special.get_filename(document.text)
-            sql, message = special.open_external_editor(filename,
-                                                          sql=document.text)
+            sql, message = special.open_external_editor(
+                filename, sql=document.text,
+                default_text=self.get_last_query() or '')
```
Comment (meeuw) [inline mycli/main.py]: can `or ''` be removed here? I think `None` should be handled correctly in `special.open_external_editor`.

actually I'm not sure about `open_external_editor`, why does it need to decide to open `sql` or `default_text`?

Comment (tsroten) [inline mycli/main.py]: Yeah, I'd like to streamline this a bit. I'll think through it.

`open_external_editor` is where the \e is removed, which `handle_editor_command` doesn't know about.

**Comment by tsroten** [comment]
@meeuw Ok, I think I've addressed your points. What do you think?

**Comment by meeuw** [comment]
:+1: :+1:

**Pull request merged**